src/System.Security.Cryptography.Xml/tests/XmlLicenseEncryptedRef.cs

```diff
@@ -0,0 +1,229 @@
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class XmlLicenseEncryptedRef : IRelDecryptor
+    {
+        List<AsymmetricAlgorithm> asymmetricKeys = new List<AsymmetricAlgorithm>();
+
+        public XmlLicenseEncryptedRef()
+        {
+        }
+
+        public void AddAsymmetricKey(AsymmetricAlgorithm key)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            asymmetricKeys.Add(key);
+        }
+
+        private static bool ArrayEqual(byte[] a, byte[] b)
+        {
+            if (a.Length != b.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < a.Length; i++)
+            {
+                if (a[i] != b[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool PublicKeysEqual(RSAParameters a, RSAParameters b)
+        {
+            return ArrayEqual(a.Exponent, b.Exponent) && ArrayEqual(a.Modulus, b.Modulus);
+        }
+
+        public Stream Decrypt(EncryptionMethod encryptionMethod, KeyInfo keyInfo, Stream toDecrypt)
+        {
+            Assert.NotNull(encryptionMethod);
+            Assert.NotNull(keyInfo);
+            Assert.NotNull(toDecrypt);
+            Assert.True(encryptionMethod.KeyAlgorithm == EncryptedXml.XmlEncAES128Url
+                     || encryptionMethod.KeyAlgorithm == EncryptedXml.XmlEncAES192Url
+                     || encryptionMethod.KeyAlgorithm == EncryptedXml.XmlEncAES256Url);
+
+            Assert.Equal(keyInfo.Count, 1);
+
+            Aes aes = null;
+
+            foreach (KeyInfoClause clause in keyInfo)
+            {
+                if (clause is KeyInfoEncryptedKey)
+                {
+                    KeyInfoEncryptedKey encryptedKeyInfo = clause as KeyInfoEncryptedKey;
+                    EncryptedKey encryptedKey = encryptedKeyInfo.EncryptedKey;
+
+                    Assert.Equal(encryptedKey.EncryptionMethod.KeyAlgorithm, EncryptedXml.XmlEncRSAOAEPUrl);
+                    Assert.Equal(encryptedKey.KeyInfo.Count, 1);
+                    Assert.NotEqual(asymmetricKeys.Count, 0);
+
+                    RSAParameters rsaParams = new RSAParameters();
+                    RSAParameters rsaInputParams = new RSAParameters();
+
+                    foreach (KeyInfoClause rsa in encryptedKey.KeyInfo)
+                    {
+                        if (rsa is RSAKeyValue)
+                        {
+                            rsaParams = (rsa as RSAKeyValue).Key.ExportParameters(false);
+                            break;
+                        }
+                        else
+                            Assert.True(false, "Invalid License - MalformedKeyInfoClause");
```

Comment:
Style: Block statements require braces when their sibling have braces, or they're close to existing braces.  (Two reasons this needs it)